shotover/src/runner.rs

```diff
@@ -171,6 +186,44 @@ impl Shotover {
     pub fn run_block(self) -> ! {
         let (trigger_shutdown_tx, trigger_shutdown_rx) = tokio::sync::watch::channel(false);
 
+        let _received_file_descriptors = if let Some(socket_path) = &self.hotreload_from_socket {
+            info!("Hot reload CLIENT mode - requesting socket handoff from existing shotover");
+
+            let client = crate::hot_reload_client::UnixSocketClient::new(socket_path.clone());
+            match self.runtime.block_on(async {
+                client
+                    .send_request(crate::hot_reload::Request::SendListeningSockets)
+                    .await
+            }) {
+                Ok(crate::hot_reload::Response::SendListeningSockets { port_to_fd }) => {
+                    info!(
+                        "Successfully received {} file descriptors for hot reload",
+                        port_to_fd.len()
+                    );
+                    for (port, fd) in &port_to_fd {
+                        info!("Received file descriptor {} for port {}", fd.0, port);
+                    }
+                    Some(port_to_fd)
+                }
+                Ok(crate::hot_reload::Response::Error(msg)) => {
+                    error!("Hot reload request failed: {}", msg);
+                    std::process::exit(1);
```

Comment:
`std::process::exit(1);` is a very strong hammer!
It will terminate the process immediately without letting any further error handling or cleanup from occurring.

We do use it in very specific locations in `runner.rs` but those locations are considered the final point of our error handling / cleanup. We should not be introducing more usages.

In this particular case, I agree that an error response to SendListeningSockets is not recoverable for the client and it should give up and shutdown.
However, we should move all of the hot reload logic (the server as well) into `Server::run` so that we can return an error from there without needing to call `std::process::exist`.